### PR TITLE
Feature/697 - PostType Model

### PR DESCRIPTION
**Pull request by CodeProKid**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ ] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
This implements the Model Layer for PostTypes. I also had to fudge around the tests a bit since I think some fields were actually not working previously.

Does this close any currently open issues?
------------------------------------------
#697 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** …

**WordPress Version:** …

### CODE REVIEW

File: src/Model/PostType.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+	/**
+	 * Initializes the object
+	 *
+	 * @access protected
+	 * @return void
+	 */
+	protected function init() {
+
+		if ( 'private' === $this->get_visibility() ) {
```
Comment (jasonbahl) [inline src/Model/PostType.php]: So, I don't see anywhere that we're restricting anything here. Is the thought that this should _all_ be public by default?

Comment (jasonbahl) [reply src/Model/PostType.php]: or wait, am I mistaken. . .would this _restrict_ it by default? and have _no_ fields set as allowed restricted fields?

**Comment by CodeProKid** [comment]
@jasonbahl made some adjustments here to prevent leaking some implementation details. Now we're checking the `edit_posts` cap on the post type to show all fields, and I'm setting the object to private if the post type isn't public.

There are some weird failing tests that we'll have to take a look at tomorrow though. Can't figure out what's going on with them.

**Comment by jasonbahl** [comment]
🔥🔥👏👏👌👌

**Comment by CodeProKid** [comment]
lol, you're too quick for me 😆

**Comment by jasonbahl** [comment]
Ah. Just saw the note about the tests. Ya. I’ll help look into it tomorrow

**Pull request merged**